### PR TITLE
Tag Grafana dashboards

### LINKED ISSUE #208: Label and organize Grafana dashboards
We currently deploy all dashboards at the 'root'. Adding labels, and putting them in folders, would be good.

### PR DESCRIPTION
**Pull request by alxf**
Fix #208, but only the tag part of the ticket.

### CODE REVIEW

File: roles/kube_prometheus/hack/fix-dashboard.py [hunk 1]
```diff
@@ -66,6 +66,13 @@ def main(dashboard_data, output, filename, title=None):
                 annotation['datasource'] = source_template
                 change = True
 
+    if tags is not None:
+        # reset initial tags list
+        dashboard['tags'] = []
```
Comment (NicolasT) [inline roles/kube_prometheus/hack/fix-dashboard.py]: `dashboard['tags'] = list(set(tag.strip() for tag in tags.split(',')))`

File: roles/kube_prometheus/Makefile [hunk 2]
```diff
@@ -15,23 +15,30 @@ clean:
 .PHONY: clean
 
 files/additional-dashboards/kube-prometheus-etcd-dashboard.yml: ../../vendor/etcd/Documentation/op-guide/grafana.json
+files/additional-dashboards/kube-prometheus-etcd-dashboard.yml: TAGS=etcd
```
Comment (NicolasT) [inline roles/kube_prometheus/Makefile]: `etcd,cluster`

File: roles/kube_prometheus/Makefile [hunk 3]
```diff
@@ -15,23 +15,30 @@ clean:
...
 
 files/additional-dashboards/kube-prometheus-elasticsearch-dashboard.yml: ../../vendor/justwatchcom/elasticsearch_exporter/examples/grafana/dashboard.json
 files/additional-dashboards/kube-prometheus-elasticsearch-dashboard.yml: TITLE=Elasticsearch
+files/additional-dashboards/kube-prometheus-elasticsearch-dashboard.yml: TAGS=elasticsearch
```
Comment (NicolasT) [inline roles/kube_prometheus/Makefile]: `elasticsearch,logging`

File: roles/kube_prometheus/Makefile [hunk 4]
```diff
@@ -15,23 +15,30 @@ clean:
...
 
 files/additional-dashboards/kube-prometheus-node-exporter-dashboard.yml: ../../vendor/rfrail3/grafana-dashboards/prometheus/node-exporter-full-old.json
 files/additional-dashboards/kube-prometheus-node-exporter-dashboard.yml: TITLE=Nodes (Detailed)
+files/additional-dashboards/kube-prometheus-node-exporter-dashboard.yml: TAGS=nodes
```
Comment (NicolasT) [inline roles/kube_prometheus/Makefile]: `nodes,cluster`

File: roles/kube_prometheus/Makefile [hunk 5]
```diff
@@ -15,23 +15,30 @@ clean:
...
 
 files/additional-dashboards/kube-prometheus-prometheus-dashboard.yml: ../../vendor/grafana/grafana/public/app/plugins/datasource/prometheus/dashboards/prometheus_2_stats.json
+files/additional-dashboards/kube-prometheus-prometheus-dashboard.yml: TAGS=dashboard,prometheus
```
Comment (NicolasT) [inline roles/kube_prometheus/Makefile]: `prometheus,metrics`

File: roles/kube_prometheus/Makefile [hunk 6]
```diff
@@ -15,23 +15,30 @@ clean:
...
 
 files/additional-dashboards/kube-prometheus-calico-dashboard.yml: ../../vendor/grafana-dashboards/3244/revisions/1/download
+files/additional-dashboards/kube-prometheus-calico-dashboard.yml: TAGS=calico
```
Comment (NicolasT) [inline roles/kube_prometheus/Makefile]: `calico,cluster,networking`

File: roles/kube_prometheus/Makefile [hunk 7]
```diff
@@ -15,23 +15,30 @@ clean:
...
 
 files/additional-dashboards/kube-prometheus-calico2-dashboard.yml: ../../vendor/zihaoyu/calico/667584956586c0b869d6ad9573ff8312efa265f1/master/reference/felix/grafana-dashboard.json
 files/additional-dashboards/kube-prometheus-calico2-dashboard.yml: TITLE=Kubernetes Calico (Alternative)
+files/additional-dashboards/kube-prometheus-calico2-dashboard.yml: TAGS=calico
```
Comment (NicolasT) [inline roles/kube_prometheus/Makefile]: `calico,cluster,networking`

File: roles/kube_prometheus/Makefile [hunk 8]
```diff
@@ -15,23 +15,30 @@ clean:
...
 
 files/additional-dashboards/kube-prometheus-nginx-ingress-dashboard.yml: ../../vendor/kubernetes/ingress-nginx/deploy/grafana/dashboards/nginx.yaml
+files/additional-dashboards/kube-prometheus-nginx-ingress-dashboard.yml: TAGS=nginx
```
Comment (NicolasT) [inline roles/kube_prometheus/Makefile]: `nginx,cluster,networking`

File: roles/kube_prometheus/Makefile [hunk 9]
```diff
@@ -15,23 +15,30 @@ clean:
...
 
-files/additional-dashboards/%.yml:
+files/additional-dashboards/%.yml: hack/fix-dashboard.py
```
Comment (NicolasT) [inline roles/kube_prometheus/Makefile]: Should this also include `Makefile` then? I seem to remember @Zempashi brought this up some time ago as well.

**Review by NicolasT — CHANGES_REQUESTED** [review]
Tag changes are suggestions. I think going beyond the name of the service being exposed (which is already part of the dashboard title anyway) is worth it.

**Comment by alxf** [comment]
@NicolasT I made all changes, but i need to reopen the pull request for bert-e.

**Pull request closed without merging**